### PR TITLE
v2 volume: add "Selective V2 Data Engine Activation" doc

### PR DESCRIPTION
**Pull request by derekbit**
#### Which issue(s) this PR fixes:

Issue Longhorn/longhorn#7015

#### What this PR does / why we need it:

- Add "Selective V2 Data Engine Activation" doc
- Rename spdk to v2-data-engine

### CODE REVIEW

**Review by innobead — DISMISSED** [review]

**Comment by innobead** [comment]
After @jillian-maroket review, we can merge this.

File: content/docs/1.6.0/v2-data-engine/quick-start.md [hunk 1]
```diff
@@ -124,33 +124,30 @@ Or, you can enable it in `Setting > General > V2 Data Engine`.
 
 ### CPU and Memory Usage
 
-When the V2 Data Engine is enabled, each instance-manager pod utilizes 1 CPU core. This high CPU usage is attributed to the `spdk_tgt` process running within each instance-manager pod. The spdk_tgt process is responsible for handling input/output (IO) operations and requires intensive polling. As a result, it consumes 100% of a dedicated CPU core to efficiently manage and process the IO requests, ensuring optimal performance and responsiveness for storage operations.
+When the V2 Data Engine is enabled, each instance-manager pod for V2 Data Engine utilizes 1 CPU core. This high CPU usage is attributed to the `spdk_tgt` process running within each instance-manager pod. spdk_tgt process is responsible for handling input/output (IO) operations and requires intensive polling. As a result, it consumes 100% of a dedicated CPU core to efficiently manage and process the IO requests, ensuring optimal performance and responsiveness for storage operations.
```
Comment (jillian-maroket) [inline content/docs/1.6.0/v2-data-engine/quick-start.md:127]:
```suggestion
When the V2 Data Engine is enabled, each Instance Manager pod for the V2 Data Engine uses 1 CPU core. The high CPU usage is caused by `spdk_tgt`, a process running in each Instance Manager pod that handles input/output (IO) operations and requires intensive polling. `spdk_tgt` consumes 100% of a dedicated CPU core to efficiently manage and process the IO requests, ensuring optimal performance and responsiveness for storage operations.
```

File: content/docs/1.6.0/references/settings.md [hunk 2]
```diff
@@ -448,7 +448,7 @@ Setting that allows you to enable the V1 Data Engine.
 
 > Default: `false`
 
-This allows users to activate the v2 data engine based on SPDK. Currently, it is in the preview phase and should not be utilized in a production environment. For more information, please refer to [V2 Data Engine (Preview Feature)](../../spdk).
+This allows users to activate the v2 data engine based on SPDK. Currently, it is in the preview phase and should not be utilized in a production environment. For more information, please refer to [V2 Data Engine (Preview Feature)](../../v2-data-engine).
```
Comment (jillian-maroket) [inline content/docs/1.6.0/references/settings.md:451]:
```suggestion
Setting that allows you to enable the V2 Data Engine, which is based on the Storage Performance Development Kit (SPDK). The V2 Data Engine is a preview feature and should not be used in production environments. For more information, see [V2 Data Engine (Preview Feature)](../../v2-data-engine).
```

File: content/docs/1.6.0/references/settings.md [hunk 3]
```diff
@@ -476,7 +476,7 @@ Number of millicpus on each node to be reserved for each instance manager pod wh
 
 > Default: `enabled`
 
-This setting allows users to enable the offline replica rebuilding for volumes using v2 data engine. For more information, please refer to [Automatic Offline Replica Rebuilding](../../spdk/automatic-offline-replica-rebuilding).
+This setting allows users to enable the offline replica rebuilding for volumes using v2 data engine. For more information, please refer to [Automatic Offline Replica Rebuilding](../../v2-data-engine/features/automatic-offline-replica-rebuilding).
```
Comment (jillian-maroket) [inline content/docs/1.6.0/references/settings.md:479]:
```suggestion
Setting that allows rebuilding of offline replicas for volumes using the V2 Data Engine. For more information, see [Automatic Offline Replica Rebuilding](../../v2-data-engine/features/automatic-offline-replica-rebuilding).
```

File: content/docs/1.6.0/references/settings.md (hunk 2 above)
Comment (jillian-maroket) [reply content/docs/1.6.0/references/settings.md:451]: The suggested setting descriptions are from [Helm Values](https://longhorn.io/docs/1.6.0/references/helm-values/).

File: content/docs/1.6.0/v2-data-engine/features/automatic-offline-replica-rebuilding.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+## Settings
+
+### Global Settings
+
+- **offline-replica-rebuilding** <br>
+
+    This setting allows users to enable the offline replica rebuilding for volumes using V2 Data Engine. The value is `enabled` by default, and available options are:
+
+    - **disabled**
+    - **enabled**
```
[the remark this answers is not in the document]
Comment (jillian-maroket) [reply content/docs/1.6.0/v2-data-engine/features/automatic-offline-replica-rebuilding.md:17]: The format and headings I am suggesting here have been implemented in [Snapshot Space Management](https://longhorn.io/docs/1.6.0/advanced-resources/snapshot-space-management/).

**Review by jillian-maroket — COMMENTED** [review]
I reviewed the v1.6.0 doc changes. Please apply the suggestions to all affected versions.

**Comment by derekbit** [comment]
@jillian-maroket The changes LGTM. Thank you.

**Comment by jillian-maroket** [comment]
@derekbit Will you apply the suggestions to the other versions as well?

**Comment by derekbit** [comment]
> @derekbit Will you apply the suggestions to the other versions as well?

Yes, I will update it later.

**Review by jillian-maroket — APPROVED** [review]
The v1.6.0 changes LGTM. Please apply the relevant suggestions to the other versions.

**Pull request merged**